pkg/identity/userid.go

```diff
@@ -0,0 +1,104 @@
+package identity
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+	"github.com/tyler-smith/go-bip39"
+
+	"github.com/threefoldtech/zos/pkg/versioned"
+)
+
+// UserIdentity defines serializable strict to identify a user
+type UserIdentity struct {
+	// Seed encoded in base64
+	Mnemonic string `json:"mnemonic"`
+	// ThreebotID generated by explorer
+	ThreebotID uint64 `json:"threebotid"`
+}
+
+// UserData contains all information needed to identify someone
+type UserData struct {
```

Comment:
This struct could be avoided I think. You already have all the information required in UserIdentity. Instead of having another struct just to have access to the KeyPair, you can just add a KeyPair method on the UserIdentity struct.

Reply:
Ah, yes. Wasn't sure we could do that in Go, it's much better, for sure. I'll change that.